indirect_cxx14.h

```diff
@@ -197,9 +197,7 @@ class indirect : private detail::empty_base_optimization<A> {
     } else {
       if (std::is_copy_assignable<T>::value && !valueless_after_move() &&
           alloc_base::get() == other.alloc_base::get()) {
-        T tmp(*other);
```

Comment:
We assign rather than copy-swap this because we _just_ want to store an object in dynamically assigned memory not alter exception safety guarantees. basic_optional in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p2047r7.html does the same.